net/httpclient.go

```diff
@@ -125,9 +132,41 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 			req.Header.Set("Authorization", "Bearer "+string(b))
 		}
 	}
+	if req.Body != nil && req.Body != http.NoBody && req.ContentLength > 0 {
+		retryBuffer := skpio.NewCopyBodyStream(int(req.ContentLength), &bytes.Buffer{}, req.Body)
+		c.retryBuffers.Store(req, retryBuffer)
```

Comment:
Hm, I do not see a corresponding Delete. Looks like this will leak buffers.

Reply:
Do we need this in the client? Why not let users use CopyBodyStream like we do in proxy?

Reply:
The proxy code is about one request, but here it's about a client that could hundreds of requests and we need to get the right body for the retry. If we want to support a retry with body, we need to store it somewhere.

Reply:
I saw we can use LoadAndDelete() instead of Load(), but this would make only a single retry possible, but maybe that's good enough for now.

Reply:
> Do we need this in the client? Why not let users use CopyBodyStream like we do in proxy?

I thought it makes sense to provide it for simplicity as a user.